types/didyoumean/index.d.ts

```diff
@@ -1,17 +1,12 @@
 export = didYouMean;
 
-declare function didYouMean(str: string, list: string[], key?: string): string | string[];
+declare function didYouMean(str: string, list: string[] | Record<string, string>[], key?: string): string | string[] | null;
```

Comment:
From the [logs](https://github.com/DefinitelyTyped/DefinitelyTyped/actions/runs/10332781096/job/28604394291?pr=70017), CI fails at the lint step since `Array<T>` is the covention preferred for non-simple types, [ref](https://typescript-eslint.io/rules/array-type/). In this case, it'd be `Array<Record<string, string>>`.

Also, as mentioned, `string[]` can be removed from the return type.

```suggestion
declare function didYouMean(str: string, list: string[] | Array<Record<string, string>>, key?: string): string | null;
```